x-pack/plugins/uptime/public/components/queries/monitor_charts/monitor_charts.tsx

```diff
@@ -32,232 +32,264 @@ interface MonitorChartsProps {
-  autorefreshIsPaused,
-  autorefreshInterval,
-}: Props) => (
-  <Query
-    pollInterval={autorefreshIsPaused ? undefined : autorefreshInterval}
-    query={createGetMonitorChartsQuery}
-    variables={{ dateRangeStart, dateRangeEnd, monitorId }}
-  >
-    {({ loading, error, data }) => {
-      if (loading) {
-        return i18n.translate('xpack.uptime.monitorCharts.loadingMessage', {
-          defaultMessage: 'Loading…',
-        });
-      }
-      if (error) {
-        return i18n.translate('xpack.uptime.monitorCharts.errorMessage', {
-          values: { message: error.message },
-          defaultMessage: 'Error {message}',
-        });
-      }
+export class MonitorCharts extends React.Component<Props, MonitorChartsState> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { crosshairLocation: 0 };
+  }
 
-      // TODO: this should not exist in the UI, update the GQL resolver/schema to return
-      // an object that contains these series already shaped in the way required by the visualizations.
-      const { monitorChartsData } = data;
-      const rttWriteRequestSeries: any[] = [];
-      const rttValidateSeries: any[] = [];
-      const rttContentSeries: any[] = [];
-      const rttResponseSeries: any[] = [];
-      const rttTcpSeries: any[] = [];
-      const avgDurationSeries: any[] = [];
-      const areaRttSeries: any[] = [];
-      const downSeries: any[] = [];
-      const upSeries: any[] = [];
-      const checksSeries: any[] = [];
-      const maxRtt: any[] = [];
-      monitorChartsData.forEach(
-        ({
-          maxWriteRequest,
-          maxValidate,
-          maxContent,
-          maxResponse,
-          maxTcpRtt,
-          avgDuration,
-          maxDuration,
-          minDuration,
-          status,
-        }: any) => {
-          // We're summing these values because we need to know what the max value of the RTT
-          // fields are in order to provide an accurate domain size for the RTT combination series.
-          maxRtt.push({
-            x: maxWriteRequest.x,
-            y: maxWriteRequest.y + maxValidate.y + maxContent.y + maxResponse.y + maxTcpRtt.y,
-          });
-          // TODO: these types of computations should take place on the server and be reflected in the GQL schema
-          rttWriteRequestSeries.push(maxWriteRequest);
-          rttValidateSeries.push(maxValidate);
-          rttContentSeries.push(maxContent);
-          rttResponseSeries.push(maxResponse);
-          rttTcpSeries.push(maxTcpRtt);
-          avgDurationSeries.push(avgDuration);
-          areaRttSeries.push({ x: minDuration.x, y0: minDuration.y, y: maxDuration.y });
-          downSeries.push({ x: status.x, y: status.down });
-          upSeries.push({ x: status.x, y: status.up });
-          checksSeries.push({ x: status.x, y: status.total });
-        }
-      );
+  public render() {
+    const {
+      dateRangeStart,
+      dateRangeEnd,
+      monitorId,
+      autorefreshEnabled,
```

Comment:
Tests are failing here because this is not a prop. https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8363/JOB=kibana-ciGroup1,node=immutable/console